tests/nn_test.py

```diff
@@ -115,18 +115,6 @@ def testDtypeMatchesInput(self, dtype, fn):
     out = fn(x)
     self.assertEqual(out.dtype, dtype)
 
-  @jtu.skip_on_devices("gpu", "tpu")
```

Comment:
The issue is that I'd expect `make_jaxpr`, like `jit` and `xla_computation`, to stage out the broadcast because of omnistaging. That is, while `jnp.ones(...)` is no longer a lazy value, the broadcast primitive it binds should be staged out, and the jaxpr formed should not have a large constant.

> To demonstrate it's fine, here's the output of

I suggest making a test out of this "to demonstrate it's fine" demonstration! As I said, I'd expect `make_jaxpr` to keep working (not with the same mechanism as before but still avoiding the large constant in this test). Is your demonstration showing that `xla_computation` avoids staging out the large constant, but `make_jaxpr` does not?

Reply:
To say the same thing in yet another way: we still want these tests to work, even though the mechanism we're relying on (omnistaging) is different than before (laziness). If these tests don't work anymore, we want to fix them (with omnistaging) rather than delete them.

Reply:
Is the issue just that these tests only work with omnistaging enabled now? If so, I recommend skipping these tests when omnistaging is disabled, but keeping the tests.

Reply:
The issue is the test calls `jnp.ones((10 ** 12,))`. That's more memory than my Macbook has, so we never even get to calling `make_jaxpr` without laziness support, so it's irrelevant what `make_jaxpr` does or does not do.

But that is not what the original issue was! The original issue was about the broadcast constants being materialized from the `where` inside `elu`. But that is something that categorically cannot happen with omnistaging.

This test is testing a particular detail of the fix implementation, and that detail has now changed, not that the underlying issue was fixed. In the presence of omnistaging, we are never going to not stage out broadcasts inside a `jit` block.

Reply:
I misread the test! Sorry. Peter and I followed up offline: there is some test worth having here but not the original one.

Reply:
Peter added new versions of these tests in #6012.